### ISSUE TITLE
Questions about the code

### ISSUE THREAD

**Issue by panicwq** [GuYuc/WS-DAN.PyTorch#21]
This is a piece of code in the wsdan.py program. 
According to my understanding, **M**(32) and **p**([1,32]) of np.random.choice() are different sizes, but why is there no error?

`        if self.training:`
`            attention_map = []`
`            for i in range(batch_size):`
`                attention_weights = torch.sqrt(attention_maps[i].sum(dim=(1, 2)).detach() + EPSILON)`
`                attention_weights = F.normalize(attention_weights, p=1, dim=0)`
`                k_index = np.random.choice(self.M, 2, p=attention_weights.cpu().numpy())`
`                attention_map.append(attention_maps[i, k_index, ...])`
`            attention_map = torch.stack(attention_map)  `
`        else:`
`            attention_map = torch.mean(attention_maps, dim=1, keepdim=True) `

**Comment by GuYuc** [comment]
@panicwq The size of p is [32], not [1, 32].
